### PR TITLE
num2words: linting and -th ordinal_num fix for en

### LINKED ISSUE #15: ordinal_num for English wrong with teens
`num2words 12 --to ordinal`
produces
`12nd `
instead of expected 
`12th`

### LINKED ISSUE #16: A number of clippy lints all over the code
Nothing serious, really, but some places can really be better with fixing those lints, like

```
warning: `to_string` applied to a type that implements `Display` in `eprintln!` args
   --> src/bin/bin.rs:120:51
    |
120 |             Err(err) => eprintln!("Error: {}", err.to_string()),
    |                                                   ^^^^^^^^^^^^ help: remove this
```

if it still is there, will get to it after adding Ukrainian

### PR DESCRIPTION
**Pull request by pavloslav**
Closes #15, closes #16

### CODE REVIEW

File: src/num2words.rs [hunk 1]
```diff
@@ -319,13 +319,11 @@ mod tests {
 
     #[test]
     fn test_string_not_valid() {
-        match Num2Words::parse("not a number") {
-            Some(_) => assert!(false),
-            None => assert!(true),
+        if Num2Words::parse("not a number").is_some() {
```
Comment (Ballasi) [inline src/num2words.rs:322]: LGTM on most side here, I am just not a huge fan of the idea of using `panic` which will 1. stop the unit tests here, without running the other ones and 2. don't report anything if it works.

The following should do a better job:
```rust
assert!(Num2Words::parse("not a number").is_none());
assert!(Num2Words::parse("NAN").is_none());
```

**Review by Ballasi — COMMENTED** [review]
Just a little thing on the unit test, the rest looks good to me!

File: src/num2words.rs (hunk 1 above)
Comment (pavloslav) [reply src/num2words.rs:322]: I just fixed https://rust-lang.github.io/rust-clippy/rust-1.74.0/index.html#/assertions_on_constants
I agree your version is better.

Comment (Ballasi) [reply src/num2words.rs:322]: Related comment for tracking: https://github.com/Ballasi/num2words/commit/a109dbcb0914dc897f17d9b709a30206bf1d9bd5#r133143809

**Pull request merged**